packages/hoppscotch-common/src/helpers/teams/TeamCollectionAdapter.ts

```diff
@@ -324,10 +324,10 @@ export default class NewTeamCollectionAdapter {
 
       if (!parentCollection) return
 
+      // Prevent adding child collections to a collection that has not been expanded yet incoming from GQL subscription, during import, etc
+      // Hence, add entries to the pre-existing list without setting 'children' if it is `null'
       if (parentCollection.children !== null) {
         parentCollection.children.push(collection)
-      } else {
-        parentCollection.children = [collection]
       }
```

Comment:
There was an issue specific to team workspaces on cloud, where requests were lost while importing except for the last level. This also extends to the case where child collections/requests are being created for a collection in the non-expanded state having pre-existing children and expanding will reveal only the above additions. There was a change in behaviour in the `teamCollectionAdded` GQL subscription on cloud compared to SH where it was fired post-collection create actions. Child collections at all levels get populated under the imported collection or while creating child collections they get synced straightaway via the subscription. This has been patched at the tree level ensuring collection entries received via GQL subscriptions for a parent collection that is not expanded yet doesn't get populated.